### PR TITLE
timeout-minutes, ff-timeout-minutes and max-parallel

### PR DESCRIPTION
**Pull request by Tibor17**
Added `timeout-minutes`, `ff-timeout-minutes` and `max-parallel`.
This can be an update in V2 or a new V3.

### CODE REVIEW

**Comment by Tibor17** [comment]
@slawekjaranowski
Would you pls check this? Thx

File: .github/workflows/maven-verify.yml [hunk 1]
```diff
@@ -170,6 +189,7 @@ jobs:
         jdk: ${{ fromJSON( inputs.jdk-matrix ) }}
         distribution: ${{ fromJSON( inputs.jdk-distribution-matrix ) }}
         exclude: ${{ fromJSON( inputs.matrix-exclude ) }}
+      max-parallel: ${{ inputs.max-parallel }}
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:192]: Please fix indent

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
Please not introduce next version. 
Target is to have one version, more only for migration.
Next version will be needed only when we break compatibility.

With next version we should manually update all repository where shared actions are used.

This change is ok for all.

File: .github/workflows/maven-verify.yml (hunk 1 above)
Comment (Tibor17) [reply .github/workflows/maven-verify.yml:192]: But the indent is right, the problem is that GH shows you only 3 lines above, these three lines are nested ones. It looks like this:

```
    strategy:
      fail-fast: ${{ inputs.verify-fail-fast }}
      matrix:
        os: ${{ fromJSON( inputs.os-matrix ) }}
        jdk: ${{ fromJSON( inputs.jdk-matrix ) }}
        distribution: ${{ fromJSON( inputs.jdk-distribution-matrix ) }}
        exclude: ${{ fromJSON( inputs.matrix-exclude ) }}
      **max-parallel: ${{ inputs.max-parallel }}**

    steps:
      - name: Should run
        id: should-run
```

Comment (slawekjaranowski) [reply .github/workflows/maven-verify.yml:192]: 😄

**Review by slawekjaranowski — APPROVED** [review]

**Comment by Tibor17** [comment]
@slawekjaranowski 
Thx, so i will push it to the `v2` branch.

**Pull request merged**